### PR TITLE
Internationalize the API

### PR DESCRIPTION
**Pull request by fpagnoux**
Connected to #76.

Depends on https://github.com/openfisca/openfisca-core/pull/436
Depends on #81 

* Make the api compatible with other countries than `france` and `tunisia`

Some of the API-related business is done in the `scenarios`. I'll generalise it later in a more complex PR, but this is enough to make the api work with `senegal` (provided that the scenario has been copy pasted and adapted from `france`).

--------------

Todo  :
- [x] Update the `setup.py` requirement once openfisca/openfisca-core#454 is merged

### CODE REVIEW

File: openfisca_web_api/model.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def get_cached_or_new_decomposition_json(tax_benefit_system, xml_file_name = Non
         xml_file_name = tax_benefit_system.DEFAULT_DECOMP_FILE
     global decomposition_json_by_file_path_cache
     decomposition_json = decomposition_json_by_file_path_cache.get(xml_file_name)
-    if decomposition_json is None:
+    if decomposition_json is None and hasattr(tax_benefit_system, 'DECOMP_DIR'):
```
Comment (cbenz) [inline openfisca_web_api/model.py]: Why would be the `get_cached_or_new_decomposition_json` function called if there is no `DECOMP_DIR` defined in the `tax_benefit_system`?

[environment.py](https://github.com/openfisca/openfisca-web-api/blob/7ae69fec1f9e8c76d5c41068878f402d22a0955c/openfisca_web_api/environment.py#L147-L148) calls this function checking for `DEFAULT_DECOMP_FILE`.

Where was the error triggered?

**Comment by fpagnoux** [comment]
To reproduce:
`core#4.3.2`
`senegal#0.3.1`
`web-api#580af38c1a3442b395dd63124bce4bafa2c71987 `

Just try to serve the api with `paster serve api/api_config.ini` from the `senegal` repo.

**Comment by fpagnoux** [comment]
Stack trace:
```sh
  File "/Users/fpagnoux/dev/mes-aides/openfisca/openfisca/web-api/openfisca_web_api/application.py", line 71, in make_app
    environment.load_environment(global_conf, app_conf)
  File "/Users/fpagnoux/dev/mes-aides/openfisca/openfisca/web-api/openfisca_web_api/environment.py", line 148, in load_environment
    model.get_cached_or_new_decomposition_json(tax_benefit_system)
  File "/Users/fpagnoux/dev/mes-aides/openfisca/openfisca/web-api/openfisca_web_api/model.py", line 42, in get_cached_or_new_decomposition_json
    xml_file_path = os.path.join(tax_benefit_system.DECOMP_DIR, xml_file_name)
AttributeError: 'SenegalTaxBenefitSystem' object has no attribute 'DECOMP_DIR'
```

**Comment by fpagnoux** [comment]
Since https://github.com/openfisca/openfisca-core/pull/397, we put `DEFAULT_DECOMP_FILE` as a general attribute of a `TaxBenefitSystem`.
This is what mislead [this condition](https://github.com/openfisca/openfisca-web-api/blob/580af38c1a3442b395dd63124bce4bafa2c71987/openfisca_web_api/environment.py#L147).

**Comment by fpagnoux** [comment]
See #81

**Comment by cbenz** [comment]
@fpagnoux I cannot ask you to review since you're the author of the PR.

What should I do? Recreate a new PR replacing this one, or asking you to review in normal comments?

**Comment by fpagnoux** [comment]
Asking a review in regular comments seems good to me in this case.

In which context did you encounter the error ? With which route ?I don't remember running into something about a description.

**Comment by fpagnoux** [comment]
Tested and approved with `senegal`.
Seems GTM for me.

**Comment by cbenz** [comment]
About `description`s as said in the Changelog, this is about the `parameters` endpoint, which returned empty strings when no description was set; there was no exception.

This did not happen with France and I saw it on Senegal.

---

In terms of practices used in the API:

good:
```json
{}
```

bad:
```json
{
  "description": null
}
```

ugly:
```json
{
  "description": ""
}
```

**Pull request merged**